### ISSUE TITLE
Regex for message substitutions in testing system could match code

### ISSUE THREAD

**Issue by JKillian** [palantir/tslint#967]
The following regex is used to match message substitution lines in test files:

```
/^\[([\w\-\_]+?)]: \s*(.+?)\s*$/
```

This is intended to match lines like this:

```
[tabs]: Tab indentation required instead of space indentation
```

However, it could get a false positive on code like this (the third line):

```
var foo = {
bar: "2",
[variable]: 52
};
```

This is a relatively minor problem, because in almost all cases like this, there will be indentation and thus leading spaces/tabs and the regex won't match.

One possible solution is to only check for message substitution lines at the very end of test files.

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is being deprecated** and no longer accepting pull requests for major new changes or features. See #4534. 😱

If you'd like to see this change implemented, you have two choices:
* **Recommended:** Check if this is available in [ESLint](https://eslint.org) + [typescript-eslint](https://typescript-eslint.io) ✅
* _Not Recommended: Fork TSLint locally_ 🤷‍♂️

👋 It was a pleasure open sourcing with you!

_If you believe this message was posted here in error, please comment so we can re-open the issue!_

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
